src/plugins/memoryvalue/testmod_memoryvalue.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file
+ * @brief Tests the memory value plugin
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdbconfig.h>
+#include <kdbtypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tests_plugin.h>
+
+// the boolean parameter is needed, because we need unsigned longs as expected values, signed longs are too short thus no way to check for
+// invalid values with -1
+static void test_memoryvalue (const char * memoryvaluestring, kdb_unsigned_long_long_t memoryvalueexpected, bool errorExpected)
+{
+	Key * parentKey = keyNew ("user/tests/memoryvalue", KEY_END);
+	Key * hexkey =
+		keyNew ("user/test/memoryvalue/testmemval", KEY_VALUE, memoryvaluestring, KEY_META, "check/memoryvalue", "any", KEY_END);
+	KeySet * conf = ksNew (0, KS_END);
+	KeySet * ks = ksNew (20, KS_END);
+	// KeySet * ksGet = ks;
+
+	ksAppendKey (ks, hexkey);
+
+	PLUGIN_OPEN ("memoryvalue");
+
+	succeed_if ((plugin->kdbSet (plugin, ks, parentKey) >= 1 || errorExpected), "kdbSet did not succeed");
+
+	succeed_if ((plugin->kdbGet (plugin, ks, parentKey) >= 1 || errorExpected), "kdbGet did not succeed");
```

Comment:
You should call `kdbGet` then check that the value was correctly transformed (or an error occurred as expected). After that you should call `kdbSet` and check that the value was correctly restored.

You could also add some tests where the value is changed in between `kdbGet` and `kdbSet` and check that `kdbSet` doesn't overwrite the change.